### DISCUSSION TITLE
Is it possible to add Pandas AI to existing chatbot providers (Azure Bot Service, Rasa, Botpress, etc) ?

### DISCUSSION THREAD

**Discussion by al-yakubovich** [sinaptik-ai/pandas-ai#196] [Q&A]
I did not find a solution to get Pandas AI working with existing providers because they are more focused on textual output

**Comment by gventuri** [comment]
Hey @al-yakubovich, thanks a lot for sharing. I'm not sure what you have in mind. Guess you can implement PandasAI in existing chatbot providers. Would you mind sharing more info about what you have in mind? You mean displaying the charts?

**Reply by al-yakubovich** [reply]
Hi @gventuri . I am thinking about the possibility of creating a chat experience with Pandas AI. Something like ChatGPT Code Interpreter, where an user can upload files to chat:  
![image](https://github.com/gventuri/pandas-ai/assets/12928778/6e5d564a-e0a9-4051-bd10-a3343f6ee55b)
and next interact with Code Interpreter so you can get visual output:
![image](https://github.com/gventuri/pandas-ai/assets/12928778/f32d8d9e-26dc-4cfd-abab-be7832ee3a9e)
